backends/dpdk/dpdkArch.h

```diff
@@ -987,6 +988,56 @@ class ElimHeaderCopy : public Transform {
     const IR::Node* postorder(IR::Member* m) override;
 };
 
+/**
+ * Flatten header union into its individual element.
+ * For ex:
+ * header_union U {
+ *     Hdr1 h1;
+ *     Hdr2 h2;
+ * }
+ *
+ * struct Headers {
+ *     Hdr1 h1;
+ *     U u;
+ * }
+ *
+ * is replaced by
+ * struct  Headers {
```

Comment:
What is the union is a standalone variable?
Just doing this change won't preserve accurately the semantics of unions.
In particular, setting a union field to valid should set all other fields to invalid.

Reply:
>What is the union is a standalone variable?

Standalone union variable should be handled by the "Declaration_Variable" visitor. Do you see any other possibilities?

Thanks for pointing the semantics w.r.t validity bit of header unions. So, every assignment to a header union element must be converted to a sequence of statements involving validity bit modifications.
Going over the specification again, I have found the following statements which need to be handled. Please let me know if anything is missing.

```
1) U u;
H1 my_h1 = { 8w0 }; 
u.h1 = my_h1;       
translates to
 if (my_h1.isValid()) {
     u_h1.setValid();
	 ///all other elements of union set to invalid
	 u_hi = my_h1;
 } else{
     u_h1.setInvalid();
 }
 
2) u.h1 = { 16w1}
translates to
u_h1.setValid();
  ///all other elements of union set to invalid
   u_h1 = {16w1};
   
3) u.h1.setValid()
translates to 
u_h1.setValid();
  ///all other elements of union set to invalid
  
4) U u1, u2;
   u1 = u2 
translates to
if (u2.isValid){
    for elements in u1{
       if element is valid{
	     copy element and set validity bit
		 set all other elements invalid and break;	   
	   }	      
	}
} else {
    ///all elements of union set to invalid
}

5) Comparison of header unions
if (u == u1)
...
translates to
tmp = u == u1;
if(u.isInvalid && u1.isInvalid)
    tmp = 1;
else if ((u.isValid && u1.isInvalid) || (u.isInvalid && u1.isValid))
    tmp = 0;
else {
//both are valid, check individual fields
//in both unions the same field is valid, and the values of the valid fields are equal as headers).
   tmp = 1;
}
if (tmp)
....
```